src/engine/sparqlExpressions/StringExpressions.cpp

```diff
@@ -200,25 +200,33 @@ using StrlenExpression =
 
 // LCASE
 [[maybe_unused]] auto lowercaseImpl =
-    [](std::optional<std::string> input) -> IdOrLiteralOrIri {
+    [](std::optional<ad_utility::triple_component::Literal> input)
+    -> IdOrLiteralOrIri {
   if (!input.has_value()) {
     return Id::makeUndefined();
   } else {
-    return toLiteral(ad_utility::utf8ToLower(input.value()));
+    auto new_content =
+        ad_utility::utf8ToLower(asStringViewUnsafe(input.value().getContent()));
+    input.value().replaceContentWithSameLength(new_content);
+    return LiteralOrIri(std::move(input.value()));
   }
 };
-using LowercaseExpression = StringExpressionImpl<1, decltype(lowercaseImpl)>;
+using LowercaseExpression = LiteralExpressionImpl<1, decltype(lowercaseImpl)>;
 
 // UCASE
 [[maybe_unused]] auto uppercaseImpl =
-    [](std::optional<std::string> input) -> IdOrLiteralOrIri {
+    [](std::optional<ad_utility::triple_component::Literal> input)
+    -> IdOrLiteralOrIri {
   if (!input.has_value()) {
     return Id::makeUndefined();
   } else {
-    return toLiteral(ad_utility::utf8ToUpper(input.value()));
+    auto new_content =
+        ad_utility::utf8ToUpper(asStringViewUnsafe(input.value().getContent()));
+    input.value().replaceContentWithSameLength(new_content);
+    return LiteralOrIri(std::move(input.value()));
```

Comment:
This is a lot of code duplication, you can template a lambda like this

```
template <auto toLowerOrToUpper>
auto upperOrLowerCaseImpl = [](......) -> IdOrLitOrIri {
 ... // same code as before
 auto newContent = std::invoke(toLowerOrToUpper(asStringViewUnsafe(...));
...
};
auto uppercaseImpl = upperOrLowerCaseImpl<&ad_utility::utf8ToUpper>;
// same for upperCaseImpl.
}
```
If there is a problem with the direct passing of the function pointers as template arguments (that might happen if the functions are templated or overloaded), then wrap them in a small lambda like
`constexpr auto toLower = [](const auto& blubb) {return utf8ToUpper(blubb)};`
And pass that lambda (the lambda, not its type) as the template argument.